src/Application.vala

```diff
@@ -110,8 +110,6 @@ public class Terminal.Application : Gtk.Application {
             if (working_directory != "\0") {
                 Environment.set_current_dir (working_directory); // will be sent via platform-data
             }
-
-            options.remove ("working-directory");
```

Comment:
instead of keeping "working-directory", add "new-tab" to the options, the working directory is sent via platform data.

Reply:
I now add "new-tab" to the options if "-w" option is present.